VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseDiscussionResponsesAdapter.java

```diff
@@ -219,7 +219,12 @@ private void bindViewHolderToResponseRow(DiscussionResponseViewHolder holder, fi
         holder.addCommentLayout.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                router.showCourseDiscussionComments(context, comment);
+                if (0 == comment.getChildren().size()) {
```

Comment:
This can be replaced with `comment.getChildren().isEmpty()` to make it slightly more intuitive.

Also, I noticed that the `getChildren()` can technically return `null`. It would be better to guarantee a non-null return value so that we don't need to use any null checks for handling the (unlikely) case whether this key is not included in the response (something like `return children == null ? Collections.EMPTY_LIST : children;`.